kythe/typescript/indexer.ts

```diff
@@ -660,10 +662,72 @@ class StandardIndexerContext implements IndexerHost {
   };
 }
 
-type ImportVNameSet = {
-  [where in 'type' | 'value']?:
-      {local: Readonly<VName>, remote: Readonly<VName>}
-};
+const RE_FIRST_NON_WS = /\S|$/;
+const RE_NEWLINE = /\r?\n/;
+const MAX_MS_TEXT_LENGTH = 1_000;
+/**
+ * Formats a MarkedSource text component by
+ * - stripping the text to its first MAX_MS_TEXT_LENGTH characters
+ * - trimming the text
+ * - stripping the leading whitespace of each line in multi-line string by the
+ *   shortest non-zero whitespace length. For example,
+ *   [
+ *       1,
+ *       2,
+ *     ]
+ *   becomes
+ *   [
+ *     1,
+ *     2,
+ *   ]
+ */
+function fmtMarkedSource(s: string) {
+  if (s.search(RE_FIRST_NON_WS) === s.length) {
+    // String is all whitespace, keep as-is
+    return s;
+  }
+  let is_chopped = false;
+  if (s.length > MAX_MS_TEXT_LENGTH) {
+    // Trim left first to pick up more chars before chopping the string
+    s = s.trimLeft();
+    s = s.substring(0, MAX_MS_TEXT_LENGTH);
+    is_chopped = true;
+  }
+  s = s.replace('\t', '    ');  // normalize tabs for display
```

Comment:
.replace(/\t/g, '')
or you will only replace the first tab

Reply:
thanks!